Reference/Security/auto-linking/index-v9.md

```diff
@@ -0,0 +1,229 @@
+  * When specifying this view it is 100% up to your angular view and affiliated angular controller to perform all required logic. To get started, the easiest way is to copy what the [default angular view](https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web.UI.Client/src/views/components/application/umb-login.html#L126-L140) does and then implement your angular controller to do what the [default controller](https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web.UI.Client/src/common/directives/components/application/umblogin.directive.js#L48) does.
+
+To register this configuration class, you can call the following from your `startup.cs`:
+```cs
+services.ConfigureOptions<OpenIdConnectBackOfficeExternalLoginProviderOptions>();
+```
+
+We recommend to create an extension method on the `IUmbracoBuilder`, to add the Open Id Connect Authentication, like this
+
+```cs
+public static IUmbracoBuilder AddOpenIdConnectAuthentication(this IUmbracoBuilder builder)
+{
+    builder.AddBackOfficeExternalLogins(logins =>
+    {
+        logins.AddBackOfficeLogin(
+            backOfficeAuthenticationBuilder =>
+            {
+                backOfficeAuthenticationBuilder.AddOpenIdConnect(
+                    // The scheme must be set with this method to work for the back office
+                    backOfficeAuthenticationBuilder.SchemeForBackOffice(OpenIdConnectBackOfficeExternalLoginProviderOptions.SchemaName),
+                    options =>
+                    {
+                        // use cookies
+                        options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                        // pass configured options along
+                        options.Authority = "https://localhost:5000";
+                        options.ClientId = "YOURCLIENTID";
+                        options.ClientSecret = "YOURCLIENTSECRET";
+                        // Use the authorization code flow
+                        options.ResponseType = OpenIdConnectResponseType.Code;
+                        options.AuthenticationMethod = OpenIdConnectRedirectBehavior.RedirectGet;
+                        // map claims
+                        options.TokenValidationParameters.NameClaimType = "name";
+                        options.TokenValidationParameters.RoleClaimType = "role";
+
+
+                        options.RequireHttpsMetadata = true;
+                        options.GetClaimsFromUserInfoEndpoint = true;
+                        options.SaveTokens = true;
+                        // add scopes
+                        options.Scope.Add("openid");
+                        options.Scope.Add("email");
+                        options.Scope.Add("roles");
+                        options.UsePkce = true;
+                    });
+            });
+    });
+    return builder;
+}
+```
+
+Finally this extension can also be called from the `Startup.cs`.
+
```

Comment:
It might make it more obvious to people reading this if we have a code example here.

Finally this extension can also be called from the `Startup.cs` like the example below:
```cs
services.AddUmbraco(_env, _config)
    .AddBackOffice()
    .AddWebsite()
    .AddComposers()
    .AddOpenIdConnectAuthentication()
    .Build();
```